components/Graph/Graph.tsx

```diff
@@ -160,6 +179,14 @@ export default class Graph {
     this.links = this.links.filter((link: Link): boolean => {
       return link.source !== source || link.target !== target 
     }) 
+    this.setInitialized(false)
+  }
+
+
+  getNodeById(id: string): Node | undefined {
+    return this.nodes.find((node: Node): boolean => {
+      return node.id === id 
+    }) 
```

Comment:
Please when dealing with nullable types always please prefer `null`. `undefined` is for unexpected nullable values while `null` is for intentional ones.

Reply:
Can't since this.nodes.find returns Node | undefined